backend/networkTree.js

```diff
@@ -0,0 +1,149 @@
+class Queue {
+
+    constructor() { this.list = []; }
+
+    isEmpty() { return this.list.length === 0; }
+    enqueue(elem) { this.list.push(elem); }
+    dequeue() { if(!this.isEmpty()) { return this.list.shift(); } }
+}
+
+class networkNode {
+    constructor(socketID, maxClients) {
+        this.socketID = socketID;
+        this.maxClients = maxClients;
+    }
+}
+
+class networkTree {
+
+    constructor(socketID, maxClients) {
+        this.node = new networkNode(socketID, maxClients);
+        this.subNodes = [];
+        this.transferringNodes = []; // nodes (& their children) which are in process of reconnecting
```

Comment:
As per the comment, we could rename this to reconnectingNodes. What do you think?

Reply:
Initially I thought that since nodes are "transferring" from one part of the tree to another, name seemed fine. But yeah, reconnectingNodes seems better.